File: app/src/main/java/com/gee12/mytetroid/helpers/IStorageInfoProvider.java
package com.gee12.mytetroid.helpers;

import com.gee12.mytetroid.model.Version;

public interface IStorageInfoProvider {
    void calcCounters();

    Version getFormatVersion();

    int getNodesCount();

    int getCryptedNodesCount();

    int getRecordsCount();

    int getCryptedRecordsCount();

    int getFilesCount();

    int getTagsCount();

    int getUniqueTagsCount();

    int getAuthorsCount();

    int getMaxSubnodesCount();

    int getMaxDepthLevel();
}

File: app/src/main/java/com/gee12/mytetroid/helpers/IStorageInfoProvider.kt
package com.gee12.mytetroid.helpers

import com.gee12.mytetroid.model.Version

interface IStorageInfoProvider {

    var formatVersion: Version?
    var nodesCount: Int
    var cryptedNodesCount: Int
    var recordsCount: Int
    var cryptedRecordsCount: Int
    var filesCount: Int
    var tagsCount: Int
    var uniqueTagsCount: Int
    var authorsCount: Int
    var maxSubnodesCount: Int
    var maxDepthLevel: Int

    fun calcCounters()

}